Bottom: 9d1a99c3a4dfd91e500115a2666d1b4ab7d5e50b
Top:    c2b1d3ec6e309b32af330474fcff24e4431fb31e
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 14:54:50 +0800

autofs-5.0.7 - add initialization of bind_result in-do_sasl_bind()

There is an unlikley code path where bind_result could be used uninitialized
so initialize it so it isn't incorrectly used if it has rubish in it.


---

diff --git a/modules/cyrus-sasl.c b/modules/cyrus-sasl.c
index b456333..68f9242 100644
--- a/modules/cyrus-sasl.c
+++ b/modules/cyrus-sasl.c
@@ -210,7 +210,7 @@ int
 do_sasl_bind(unsigned logopt, LDAP *ld, sasl_conn_t *conn, const char **clientout,
 	     unsigned int *clientoutlen, const char *auth_mech, int sasl_result)
 {
-	int ret, msgid, bind_result;
+	int ret, msgid, bind_result = LDAP_OTHER;
 	struct berval client_cred, *server_cred, temp_cred;
 	LDAPMessage *results;
 	int have_data, expected_data;
